controllers/openstackipset_controller.go

```diff
@@ -246,6 +245,199 @@ func (r *OpenStackIPSetReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 func (r *OpenStackIPSetReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&ospdirectorv1beta1.OpenStackIPSet{}).
-		//		Owns(&corev1.ConfigMap{}).
 		Complete(r)
 }
+
+func (r *OpenStackIPSetReconciler) removeIP(instance *ospdirectorv1beta1.OpenStackIPSet, removedHosts []string) {
+
+	// remove entry from OSIPSet status
+	for _, hostname := range removedHosts {
+		delete(instance.Status.HostIPs, hostname)
+		r.Log.Info(fmt.Sprintf("OSP Hostname %s removed from OSIPSet %s, new status %v", hostname, instance.Name, instance.Status.HostIPs))
+	}
+
+}
+
+func (r *OpenStackIPSetReconciler) syncDeletedIPs(instance *ospdirectorv1beta1.OpenStackIPSet) error {
+
+	allActiveIPSetHosts := make([]string, len(instance.Status.HostIPs))
+	for hostname := range instance.Status.HostIPs {
+		allActiveIPSetHosts = append(allActiveIPSetHosts, hostname)
+	}
+
+	// mark entry in OSNet status as deleted
+	for _, netName := range instance.Spec.Networks {
+		network := &ospdirectorv1beta1.OpenStackNet{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: netName, Namespace: instance.Namespace}, network)
+		if err != nil {
+			if k8s_errors.IsNotFound(err) {
+				r.Log.Info(fmt.Sprintf("OpenStackNet named %s not found!", netName))
+			}
+			// Error reading the object - requeue the request.
+			return err
+		}
+
+		// mark the reservation for the hosts as deleted
+		for index, reservation := range network.Status.RoleReservations[instance.Spec.RoleName].Reservations {
+
+			// set OSNet deleted flag for non exiting nodes in the ipset
+			reservationDeleted := !common.StringInSlice(reservation.Hostname, allActiveIPSetHosts)
+
+			// mark OSNet host entry as deleted
+			network.Status.RoleReservations[instance.Spec.RoleName].Reservations[index].Deleted = reservationDeleted
+
+			r.Log.Info(fmt.Sprintf("Update network status of host %s - net %s - %v", reservation.Hostname, network.Name, network.Status.RoleReservations[instance.Spec.RoleName].Reservations[index].Deleted))
+		}
+
+		// update status of OSNet
+		err = r.Client.Status().Update(context.TODO(), network)
+		if err != nil {
+			r.Log.Error(err, "Failed to update OpenStackNet status %v")
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (r *OpenStackIPSetReconciler) addIP(instance *ospdirectorv1beta1.OpenStackIPSet, newHosts common.HostnameList) error {
+
+	for _, k := range newHosts {
+		//hostRef := k.HostRef
+		hostname := k.Hostname
+
+		hostNetworkIPs := map[string]string{}
+
+		// iterate over the requested Networks
+		for _, netName := range instance.Spec.Networks {
+
+			network := &ospdirectorv1beta1.OpenStackNet{}
+			err := r.Client.Get(context.TODO(), types.NamespacedName{Name: netName, Namespace: instance.Namespace}, network)
+			if err != nil {
+				if k8s_errors.IsNotFound(err) {
+					r.Log.Info(fmt.Sprintf("OpenStackNet named %s not found!", netName))
```

Comment:
Same question here as line 274.  Or perhaps it would be better to catch the `NotFound` error in the caller and return a requeue request at X seconds?

Reply:
indeed thats better